src/wallet.js

```diff
@@ -337,14 +339,23 @@ const wallet = {
         } else {
           // If it's the last page, we update the storage and call the next loadAddress (if needed)
           // This is all done on updateHistoryData
+          let oldStore = _.clone(storage.store);
+          if (store) {
+            storage.setStore(store);
+          }
           const data = this.getWalletData();
           const historyTransactions = 'historyTransactions' in data ? data['historyTransactions'] : {};
           const allTokens = 'allTokens' in data ? data['allTokens'] : [];
+          storage.setStore(oldStore);
```

Comment:
Would be good to have some comments on all these storage changes.

Reply:
Added some comments explaining why we need to set the other stores.